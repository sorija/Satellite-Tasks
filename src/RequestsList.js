import React from "react";
import Request from "./Request.js";

class RequestsList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentPage: 1
    };
  }
  handlePageChange(pageNum) {
    return () => this.setState({ currentPage: pageNum });
  }
  divideRequests = () => {
    let startIndex = (this.state.currentPage - 1) * 5;
    let endIndex = this.state.currentPage * 5;
    return this.props.requests.slice(startIndex, endIndex);
  };

  render() {
    const { currentPage } = this.state;
    // number of pages is equal to number of requests divided by limit of requests per page and rounded up
    const length = this.props.requests.length / 5;
    const pagination = Array.from(
      { length: Math.ceil(length) },
      (_, i) => i + 1
    );
    return (
      <div>
        <ul>
          {this.divideRequests().map(request => (
            <li key={request.id}>
              <Request {...request} />
            </li>
          ))}
        </ul>
        <footer>
          <div>
            <button
              type="button"
              disabled={currentPage === 1}
              onClick={this.handlePageChange(1)}
            >
              &laquo;
            </button>
            <button
              type="button"
              disabled={currentPage === 1}
              onClick={this.handlePageChange(currentPage - 1)}
            >
              &lt;
            </button>
            {pagination.map(page => (
              <button
                key={page}
                type="button"
                disabled={page === currentPage}
                onClick={this.handlePageChange(page)}
              >
                {page}
              </button>
            ))}
            <button
              type="button"
              disabled={currentPage === pagination.length}
              onClick={this.handlePageChange(currentPage + 1)}
            >
              &gt;
            </button>
            <button
              type="button"
              disabled={currentPage === pagination.length}
              onClick={this.handlePageChange(pagination.length)}
            >
              &raquo;
            </button>
          </div>
        </footer>
      </div>
    );
  }
}
export default RequestsList;
