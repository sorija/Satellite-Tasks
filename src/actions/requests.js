export const addRequest = (satName, longitude, latitude) => ({
  type: "ADD_NEW_REQUEST",
  satName,
  longitude,
  latitude
});

export const toggleDone = id => ({
  type: "TOGGLE_DONE",
  id
});
