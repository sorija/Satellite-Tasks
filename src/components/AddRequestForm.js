import React from "react";

export class AddRequestForm extends React.Component {
  // using local state to track changes without modifying data before submitting
  constructor(props) {
    super(props);
    this.state = {
      satName: "",
      longitude: "",
      latitude: ""
    };
  }
  validateNumImput = num => {
    return !num || num.match(/^\d+(\.\d*)?$/);
  };
  onNameChange = e => {
    // store the e.target.value in a variable or event will be nullified before the .setState() fires
    const satName = e.target.value;
    this.setState(() => ({ satName }));
  };
  onLongitudeChange = e => {
    const longitude = e.target.value;
    if (this.validateNumImput(longitude)) {
      this.setState(() => ({ longitude }));
    }
  };
  onLatitudeChange = e => {
    const latitude = e.target.value;
    if (this.validateNumImput(latitude)) {
      this.setState(() => ({ latitude }));
    }
  };
  onSubmit = e => {
    e.preventDefault();
    this.props.onSubmit({
      satName: this.state.satName,
      longitude: this.state.longitude,
      latitude: this.state.latitude
    });
  };
  render() {
    const { satName, longitude, latitude } = this.state;
    return (
      <form onSubmit={this.onSubmit}>
        <input
          type="text"
          placeholder="Satellite's Name"
          value={satName}
          onChange={this.onNameChange}
          autoFocus
        />
        <input
          type="text"
          placeholder="Longitude"
          value={longitude}
          onChange={this.onLongitudeChange}
        />
        <input
          type="text"
          placeholder="Latitude"
          value={latitude}
          onChange={this.onLatitudeChange}
        />
        <input
          disabled={!satName || !longitude || !latitude}
          type="submit"
          value="Submit"
        />
      </form>
    );
  }
}

export default AddRequestForm;
