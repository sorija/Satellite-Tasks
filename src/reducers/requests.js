const requestsDefaults = [
  {
    id: 0,
    satName: "Alderan",
    longitude: 4934.539,
    latitude: 34943.023,
    done: false
  },
  {
    id: 1,
    satName: "Bespin",
    longitude: 249809.342,
    latitude: 98342.247,
    done: false
  },
  {
    id: 2,
    satName: "Coruscant",
    longitude: 23634.453,
    latitude: 18374.909,
    done: false
  },
  {
    id: 3,
    satName: "Dagobah",
    longitude: 69529.348,
    latitude: 463290.236,
    done: false
  },
  {
    id: 4,
    satName: "Dagobah",
    longitude: 132487.843,
    latitude: 876341.432,
    done: false
  },
  {
    id: 5,
    satName: "Hoth",
    longitude: 12843.394,
    latitude: 423673.634,
    done: false
  },
  {
    id: 6,
    satName: "Kashyyyk",
    longitude: 56932.893,
    latitude: 484893.208,
    done: false
  },
  {
    id: 7,
    satName: "Naboo",
    longitude: 789644.873,
    latitude: 125423.026,
    done: false
  },
  {
    id: 8,
    satName: "Naboo",
    longitude: 402834.073,
    latitude: 984283.056,
    done: false
  },
  {
    id: 9,
    satName: "Naboo",
    longitude: 673429.233,
    latitude: 3467211.882,
    done: false
  },
  {
    id: 10,
    satName: "Tatooine",
    longitude: 356983.348,
    latitude: 153389.092,
    done: false
  },
  {
    id: 11,
    satName: "Yavin",
    longitude: 934732.342,
    latitude: 175834.037,
    done: false
  },
  {
    id: 12,
    satName: "Wobani",
    longitude: 754094.346,
    latitude: 5645343.056,
    done: false
  }
];

const requestsReducer = (state = requestsDefaults, action) => {
  switch (action.type) {
    case "ADD_NEW_REQUEST":
      return [
        ...state,
        {
          id: state.length,
          satName: action.satName,
          longitude: action.longitude,
          latitude: action.latitude,
          done: false
        }
      ];
    case "TOGGLE_DONE":
      return state.map(request => {
        if (request.id === action.id) {
          return {
            ...request,
            done: !request.done
          };
        } else {
          return request;
        }
      });
    default:
      return state;
  }
};

export default requestsReducer;
