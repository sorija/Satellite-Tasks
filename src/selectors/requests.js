const getDividedRequests = (requests, currentPage) => {
  let startIndex = (currentPage - 1) * 5;
  let endIndex = currentPage * 5;
  return requests.slice(startIndex, endIndex);
};

export default getDividedRequests;
