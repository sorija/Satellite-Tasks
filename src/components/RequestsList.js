import React from "react";
import Request from "./Request.js";
import { connect } from "react-redux";
import Pagination from "./Pagination.js";
import { toggleDone } from "../actions/requests.js";
import getFilteredRequests from "../selectors/requests.js";

class RequestsList extends React.Component {
  divideRequests = () => {
    let startIndex = (this.props.currentPage - 1) * 5;
    let endIndex = this.props.currentPage * 5;
    return this.props.requests.slice(startIndex, endIndex);
  };
  onChange = e => {
    let id = Number(e.target.value);
    this.props.toggleDone(id);
  };
  render() {
    return (
      <div>
        <ul>
          {this.divideRequests().map(request => (
            <li key={request.id}>
              <Request {...request} onChange={this.onChange} />
            </li>
          ))}
        </ul>
        <footer>
          <Pagination />
        </footer>
      </div>
    );
  }
}
const mapStateToProps = state => {
  return {
    requests: getFilteredRequests(state.requests, state.filters),
    currentPage: state.currentPage
  };
};

const mapDispatchToProps = dispatch => ({
  toggleDone: id => dispatch(toggleDone(id)),
  getFilteredRequests: (requests, filters) =>
    dispatch(getFilteredRequests(requests, filters))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(RequestsList);
