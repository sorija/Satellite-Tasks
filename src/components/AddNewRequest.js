import React from "react";
import { connect } from "react-redux";
import AddRequestForm from "./AddRequestForm.js";
import { addRequest } from "../actions/requests.js";

export class AddNewRequest extends React.Component {
  constructor(props) {
    super(props);
    this.state = { addNew: false };
  }
  handleAddClick = () => {
    this.setState({ addNew: !this.state.addNew });
  };
  onSubmit = ({ satName, longitude, latitude }) => {
    this.props.addRequest(satName, parseFloat(longitude), parseFloat(latitude));
    this.setState({ addNew: false });
  };
  render() {
    let addRequestForm;
    if (this.state.addNew) {
      addRequestForm = <AddRequestForm onSubmit={this.onSubmit} />;
    }
    return (
      <div className="add-request-container">
        <button className="add-request__btn" onClick={this.handleAddClick}>
          NEW REQUEST
        </button>
        {addRequestForm}
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  addRequest: (satName, longitude, latitude) =>
    dispatch(addRequest(satName, longitude, latitude))
});

export default connect(
  null,
  mapDispatchToProps
)(AddNewRequest);
