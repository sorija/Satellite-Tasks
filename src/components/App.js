import React, { Component } from "react";
import { Provider } from "react-redux";
import configureStore from "../configureStore.js";
import RequestsList from "./RequestsList.js";
import AddNewRequest from "./AddNewRequest.js";
import RequestsFilters from "./RequestsFilters.js";
import "../styles/App.css";

const store = configureStore();

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <div className="App">
          <header className="App-header">
            <p>Hello Space!</p>
          </header>
          <RequestsFilters />
          <AddNewRequest />
          <RequestsList />
        </div>
      </Provider>
    );
  }
}

export default App;
