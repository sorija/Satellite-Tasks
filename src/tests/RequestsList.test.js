import React from "react";
import { shallow } from "enzyme";
import { RequestsList } from "../components/RequestsList";
import requests from "./fixtures/requests";

test("renders RequestsList with requests", () => {
  const wrapper = shallow(<RequestsList requests={requests} />);
  expect(wrapper).toMatchSnapshot();
});

test("renders RequestsList with no request and a message", () => {
  const wrapper = shallow(<RequestsList requests={[]} />);
  expect(wrapper).toMatchSnapshot();
});
