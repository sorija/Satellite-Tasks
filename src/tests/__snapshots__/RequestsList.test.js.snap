// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RequestsList with no request and a message 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RequestsList
    requests={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <span>
            No requests
          </span>
        </div>,
        <footer>
          <Connect(Pagination) />
        </footer>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            No requests
          </span>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No requests",
          },
          "ref": null,
          "rendered": "No requests",
          "type": "span",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(Pagination) />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "footer",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <span>
              No requests
            </span>
          </div>,
          <footer>
            <Connect(Pagination) />
          </footer>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              No requests
            </span>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No requests",
            },
            "ref": null,
            "rendered": "No requests",
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(Pagination) />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "footer",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders RequestsList with requests 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RequestsList
    requests={
      Array [
        Object {
          "done": false,
          "id": 0,
          "latitude": 22222.22,
          "longitude": 111.11,
          "satName": "Name0",
        },
        Object {
          "done": true,
          "id": 1,
          "latitude": 4444.444,
          "longitude": 33333.33,
          "satName": "Name1",
        },
        Object {
          "done": false,
          "id": 2,
          "latitude": 66666.666,
          "longitude": 555.555,
          "satName": "Name2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ul>
          <li>
            <Request
              done={false}
              id={0}
              latitude={22222.22}
              longitude={111.11}
              onChange={[Function]}
              satName="Name0"
            />
          </li>
          <li>
            <Request
              done={true}
              id={1}
              latitude={4444.444}
              longitude={33333.33}
              onChange={[Function]}
              satName="Name1"
            />
          </li>
          <li>
            <Request
              done={false}
              id={2}
              latitude={66666.666}
              longitude={555.555}
              onChange={[Function]}
              satName="Name2"
            />
          </li>
        </ul>,
        <footer>
          <Connect(Pagination) />
        </footer>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <Request
                done={false}
                id={0}
                latitude={22222.22}
                longitude={111.11}
                onChange={[Function]}
                satName="Name0"
              />
            </li>,
            <li>
              <Request
                done={true}
                id={1}
                latitude={4444.444}
                longitude={33333.33}
                onChange={[Function]}
                satName="Name1"
              />
            </li>,
            <li>
              <Request
                done={false}
                id={2}
                latitude={66666.666}
                longitude={555.555}
                onChange={[Function]}
                satName="Name2"
              />
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": <Request
                done={false}
                id={0}
                latitude={22222.22}
                longitude={111.11}
                onChange={[Function]}
                satName="Name0"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "done": false,
                "id": 0,
                "latitude": 22222.22,
                "longitude": 111.11,
                "onChange": [Function],
                "satName": "Name0",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <Request
                done={true}
                id={1}
                latitude={4444.444}
                longitude={33333.33}
                onChange={[Function]}
                satName="Name1"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "done": true,
                "id": 1,
                "latitude": 4444.444,
                "longitude": 33333.33,
                "onChange": [Function],
                "satName": "Name1",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": <Request
                done={false}
                id={2}
                latitude={66666.666}
                longitude={555.555}
                onChange={[Function]}
                satName="Name2"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "done": false,
                "id": 2,
                "latitude": 66666.666,
                "longitude": 555.555,
                "onChange": [Function],
                "satName": "Name2",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(Pagination) />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "footer",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ul>
            <li>
              <Request
                done={false}
                id={0}
                latitude={22222.22}
                longitude={111.11}
                onChange={[Function]}
                satName="Name0"
              />
            </li>
            <li>
              <Request
                done={true}
                id={1}
                latitude={4444.444}
                longitude={33333.33}
                onChange={[Function]}
                satName="Name1"
              />
            </li>
            <li>
              <Request
                done={false}
                id={2}
                latitude={66666.666}
                longitude={555.555}
                onChange={[Function]}
                satName="Name2"
              />
            </li>
          </ul>,
          <footer>
            <Connect(Pagination) />
          </footer>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <Request
                  done={false}
                  id={0}
                  latitude={22222.22}
                  longitude={111.11}
                  onChange={[Function]}
                  satName="Name0"
                />
              </li>,
              <li>
                <Request
                  done={true}
                  id={1}
                  latitude={4444.444}
                  longitude={33333.33}
                  onChange={[Function]}
                  satName="Name1"
                />
              </li>,
              <li>
                <Request
                  done={false}
                  id={2}
                  latitude={66666.666}
                  longitude={555.555}
                  onChange={[Function]}
                  satName="Name2"
                />
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": <Request
                  done={false}
                  id={0}
                  latitude={22222.22}
                  longitude={111.11}
                  onChange={[Function]}
                  satName="Name0"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "done": false,
                  "id": 0,
                  "latitude": 22222.22,
                  "longitude": 111.11,
                  "onChange": [Function],
                  "satName": "Name0",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": <Request
                  done={true}
                  id={1}
                  latitude={4444.444}
                  longitude={33333.33}
                  onChange={[Function]}
                  satName="Name1"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "done": true,
                  "id": 1,
                  "latitude": 4444.444,
                  "longitude": 33333.33,
                  "onChange": [Function],
                  "satName": "Name1",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": <Request
                  done={false}
                  id={2}
                  latitude={66666.666}
                  longitude={555.555}
                  onChange={[Function]}
                  satName="Name2"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "done": false,
                  "id": 2,
                  "latitude": 66666.666,
                  "longitude": 555.555,
                  "onChange": [Function],
                  "satName": "Name2",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(Pagination) />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "footer",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
